ml_peg/analysis/bulk_crystal/phonons/analyse_phonons.py

```diff
@@ -0,0 +1,647 @@
+    """
+    try:
+        with open(file_path, "rb") as f:
+            dos_dict = pickle.load(f)
+        return dos_dict["frequency_points"], dos_dict["total_dos"]
+    except OSError as exc:
+        print(f"Failed to load DOS from {file_path}: {exc}")
+        return None
+
+
+def _load_thermal_properties(file_path: Path) -> dict[str, Any] | None:
+    """
+    Load vibrational thermodynamic properties.
+
+    Parameters
+    ----------
+    file_path
+        Path to the JSON file containing entropy, free energy, and heat
+        capacity arrays.
+
+    Returns
+    -------
+    dict[str, Any] | None
+        Mapping of thermal quantities keyed by temperature, or ``None`` if the
+        JSON file is missing.
+    """
+    try:
+        with open(file_path, encoding="utf8") as f:
+            return json.load(f)
+    except OSError as exc:
+        print(f"Failed to load thermal properties from {file_path}: {exc}")
+        return None
+
+
+def _get_mp_ids() -> list[str]:
+    """
+    Return all Materials Project IDs with reference data.
+
+    Returns
+    -------
+    list[str]
+        Sorted collection of MP identifiers discovered in the reference DFT
+        directory.
+    """
+    ref_dir = CALC_PATH / "DFT"
+    if not ref_dir.exists():
+        print(f"ERROR: Reference directory not found: {ref_dir}")
+        print(f"Expected location: {ref_dir.absolute()}")
+        return []
+
+    mp_ids = set()
+    for file_path in ref_dir.glob("mp-*_thermal_properties.json"):
+        mp_id = file_path.stem.replace("_thermal_properties", "")
+        mp_ids.add(mp_id)
+
+    return sorted(mp_ids)
+
+
+def _prettify_chemical_formula(formula: str) -> str:
+    """
+    Convert a raw chemical formula to a LaTeX-friendly string.
+
+    Parameters
+    ----------
+    formula
+        Plain-text chemical formula (e.g. ``Al2O3``).
+
+    Returns
+    -------
+    str
+        String with subscripts added so Matplotlib renders chemical species
+        correctly.
+    """
+    import re
+
+    parts = re.findall(r"([A-Z][a-z]*)(\d*)", formula)
+    pretty = ""
+    for element, count in parts:
+        if count == "":
+            pretty += element
+        else:
+            pretty += f"{element}$_{{{count}}}$"
+    return pretty
+
+
+def _build_xticks(
+    distances: list, labels: list, connections: list
+) -> tuple[list, list]:
+    """
+    Build x-axis ticks for the band-structure plot.
+
+    Parameters
+    ----------
+    distances
+        Distances along the Brillouin-path segments.
+    labels
+        High-symmetry labels for each vertex.
+    connections
+        Boolean mask describing segment connectivity.
+
+    Returns
+    -------
+    tuple[list, list]
+        Tick positions and tick labels for Matplotlib.
+    """
+    xticks, xticklabels = [], []
+    cumulative_dist, i = 0.0, 0
+    connections = [True] + connections
+
+    for seg_dist, connected in zip(distances, connections, strict=False):
```

Comment:
With `strict=False` this will stop when either of the two lists ends, even if the other is longer. Is this what you want?